internal/formats/common/spdxhelpers/to_syft_model.go

```diff
@@ -0,0 +1,373 @@
+package spdxhelpers
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/spdx/tools-golang/spdx"
+
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/internal/formats/spdx22json/model"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/linux"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+)
+
+func ToSyftModel(doc *spdx.Document2_2) (*sbom.SBOM, error) {
+	spdxIDMap := make(map[string]interface{})
+
+	collectSyftPackages(spdxIDMap, doc)
+
+	collectSyftFiles(spdxIDMap, doc)
+
+	catalog := pkg.NewCatalog()
+
+	for _, v := range spdxIDMap {
+		if p, ok := v.(*pkg.Package); ok {
+			catalog.Add(*p)
+		}
+	}
+
+	return &sbom.SBOM{
+		Artifacts: sbom.Artifacts{
+			PackageCatalog:    catalog,
+			LinuxDistribution: findSyftLinuxRelease(doc),
+		},
+		Relationships: toSyftRelationships(spdxIDMap, doc),
+	}, nil
+}
+
+func findLinuxReleaseByDocument(doc *spdx.Document2_2) *linux.Release {
+	var p *spdx.Package2_2
+
+	if r := findSpdxRelationshipByType(doc, "DESCRIBES"); r != nil {
+		if string(r.RefA.ElementRefID) == "DOCUMENT" {
+			p = findSpdxPackageByID(doc, r.RefB.ElementRefID)
+		}
+	}
+
+	if r := findSpdxRelationshipByType(doc, "DESCRIBED_BY"); r != nil {
+		if string(r.RefB.ElementRefID) == "DOCUMENT" {
+			p = findSpdxPackageByID(doc, r.RefA.ElementRefID)
+		}
+	}
+
+	var name string
+	var version string
+
+	if p != nil {
+		name = p.PackageName
+		version = p.PackageVersion
+	}
+
+	if p != nil && (name == "" || version == "") {
+		// SPDX has relationship: PACKAGE_OF
+		// and Package External Reference PACKAGE-MANAGER
+		// spec references "PACKAGE-MANAGER" but JSON schema has "PACKAGE_MANAGER"
+		ref := findSpdxReferenceByName(p, "PACKAGE-MANAGER", "PACKAGE_MANAGER")
+
+		if ref != nil {
+			parts := strings.Split(ref.Locator, "-")
+			if len(parts) > 1 {
+				name = parts[0]
+				version = parts[1]
+			}
+		}
+	}
+
+	if name != "" && version != "" {
+		return &linux.Release{
+			PrettyName: name,
+			Name:       name,
+			ID:         name,
+			IDLike:     []string{name},
+			Version:    version,
+			VersionID:  version,
+		}
+	}
+
+	return nil
+}
+
+func findLinuxReleaseByPURL(doc *spdx.Document2_2) *linux.Release {
```

Comment:
This function looks like it could be simplified some, either by using guard clauses to reduce indention:
```golang
func findLinuxReleaseByPURL(doc *spdx.Document2_2) *linux.Release {
	for _, p := range doc.Packages {
		purlValue := extractPURL(p)
		if purlValue == "" {
			continue
		}
		
		purl, err := packageurl.FromString(purlValue)
		if err != nil {
			log.Warnf("unable to parse purl: %s", purlValue)
			continue
		} 
		for _, qualifier := range purl.Qualifiers {
			if qualifier.Key != "distro" {
				continue
			}

			parts := strings.Split(qualifier.Value, "-")
			if len(parts) != 1 {
				continue
			}

			name := parts[0]
			version := parts[1]
			return &linux.Release{
				PrettyName: name,
				Name:       name,
				ID:         name,
				IDLike:     []string{name},
				Version:    version,
				VersionID:  version,
			}	
		}
	}


	return nil
}
```

...and/or function decomposition.